meta/saisanitycheck.c

```diff
@@ -5439,6 +5439,129 @@ void check_global_apis()
+    CHECK_STRUCT_SIZE(sai_attribute_t, 56);
+    CHECK_STRUCT_SIZE(sai_bfd_session_state_notification_t, 16);
+    CHECK_STRUCT_SIZE(sai_fabric_port_reachability_t, 8);
+    CHECK_STRUCT_SIZE(sai_fdb_entry_t, 24);
+    CHECK_STRUCT_SIZE(sai_fdb_event_notification_data_t, 48);
+    CHECK_STRUCT_SIZE(sai_hmac_t, 36);
+    CHECK_STRUCT_SIZE(sai_inseg_entry_t, 16);
+    CHECK_STRUCT_SIZE(sai_ip_address_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_ip_address_t, 20);
+    CHECK_STRUCT_SIZE(sai_ipmc_entry_t, 64);
+    CHECK_STRUCT_SIZE(sai_ip_prefix_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_ip_prefix_t, 36);
+    CHECK_STRUCT_SIZE(sai_ipsec_sa_status_notification_t, 16);
+    CHECK_STRUCT_SIZE(sai_json_t, 16);
+    CHECK_STRUCT_SIZE(sai_l2mc_entry_t, 64);
+    CHECK_STRUCT_SIZE(sai_latch_status_t, 2);
+    CHECK_STRUCT_SIZE(sai_map_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_map_t, 8);
+    CHECK_STRUCT_SIZE(sai_mcast_fdb_entry_t, 24);
+    CHECK_STRUCT_SIZE(sai_my_sid_entry_t, 40);
+    CHECK_STRUCT_SIZE(sai_nat_entry_data_t, 32);
+    CHECK_STRUCT_SIZE(sai_nat_entry_key_t, 16);
+    CHECK_STRUCT_SIZE(sai_nat_entry_mask_t, 16);
+    CHECK_STRUCT_SIZE(sai_nat_entry_t, 56);
+    CHECK_STRUCT_SIZE(sai_nat_event_notification_data_t, 64);
+    CHECK_STRUCT_SIZE(sai_neighbor_entry_t, 40);
+    CHECK_STRUCT_SIZE(sai_object_key_t, 64);
+    CHECK_STRUCT_SIZE(sai_object_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_port_err_status_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_port_eye_values_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_port_lane_eye_values_t, 20);
+    CHECK_STRUCT_SIZE(sai_port_lane_latch_status_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_port_lane_latch_status_t, 8);
+    CHECK_STRUCT_SIZE(sai_port_oper_status_notification_t, 16);
+    CHECK_STRUCT_SIZE(sai_prbs_rx_state_t, 8);
+    CHECK_STRUCT_SIZE(sai_qos_map_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_qos_map_params_t, 16);
+    CHECK_STRUCT_SIZE(sai_qos_map_t, 32);
+    CHECK_STRUCT_SIZE(sai_queue_deadlock_notification_data_t, 16);
+    CHECK_STRUCT_SIZE(sai_route_entry_t, 56);
+    CHECK_STRUCT_SIZE(sai_s16_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_s32_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_s32_range_t, 8);
+    CHECK_STRUCT_SIZE(sai_s8_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_segment_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_service_method_table_t, 16);
+    CHECK_STRUCT_SIZE(sai_stat_capability_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_stat_capability_t, 8);
+    CHECK_STRUCT_SIZE(sai_system_port_config_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_system_port_config_t, 24);
+    CHECK_STRUCT_SIZE(sai_timespec_t, 16);
+    CHECK_STRUCT_SIZE(sai_tlv_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_tlv_t, 40);
+    CHECK_STRUCT_SIZE(sai_u16_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_u16_range_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_u16_range_t, 4);
+    CHECK_STRUCT_SIZE(sai_u32_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_u32_range_t, 8);
+    CHECK_STRUCT_SIZE(sai_u8_list_t, 16);
+    CHECK_STRUCT_SIZE(sai_vlan_list_t, 16);
```

Comment:
All these structs are part of 
https://github.com/opencomputeproject/SAI/blob/6f9d06e9d046c701b273574bcc42e6bc34ed6eba/meta/saimetadatatypes.h#L478C3-L478C24

Imposing this check will not let us define a new structure without impacting the size of sai_attr_value_type_t.

We need to overcome this else we will NEVER be able to add new structures.

Reply:
thats the point to make it const and be backward compatible

Reply:
you can still add new structures, but less in size of current value size